com.unity.render-pipelines.core/Runtime/Lighting/ProbeVolume/ProbeVolume.hlsl

```diff
@@ -145,102 +163,218 @@ bool TryToGetPoolUVW(APVResources apvRes, float3 posWS, float3 normalWS, float3
+#endif
+
+        bakeDiffuseLighting += apvSample.L0;
+        backBakeDiffuseLighting += apvSample.L0;
+        lightingInReflDir += apvSample.L0;
+    }
+    else
+    {
+        bakeDiffuseLighting = EvaluateAmbientProbe(normalWS);
+        backBakeDiffuseLighting = EvaluateAmbientProbe(backNormalWS);
+        lightingInReflDir = -1;
+    }
 }
 
-void EvaluateAdaptiveProbeVolume(in float3 posWS, in float3 normalWS, in float3 backNormalWS, in float3 viewDir, in float2 positionSS,
-    out float3 bakeDiffuseLighting, out float3 backBakeDiffuseLighting)
+void EvaluateAdaptiveProbeVolume(in float3 posWS, in float3 normalWS, in float3 backNormalWS, in float3 viewDir,
+    in float2 positionSS, out float3 bakeDiffuseLighting, out float3 backBakeDiffuseLighting)
 {
-    APVResources apvRes = FillAPVResources();
+    bakeDiffuseLighting = float3(0.0, 0.0, 0.0);
+    backBakeDiffuseLighting = float3(0.0, 0.0, 0.0);
 
-    // Bit of an hack to apply noise at sampling location to hide seams. Ideally we should run this only when there is a seam, but detecting that would be costly.
     if (_PVSamplingNoise > 0)
     {
-        float3x3 orthoBasis = GetLocalFrame(normalWS);
-
         float noise1D_0 = (InterleavedGradientNoise(positionSS, 0) * 2.0f - 1.0f) * _PVSamplingNoise;
-        float noise1D_1 = (InterleavedGradientNoise(positionSS, 1) * 2.0f - 1.0f) * _PVSamplingNoise;
-        posWS += orthoBasis[0] * noise1D_1 + noise1D_0 * orthoBasis[1];
+        posWS += noise1D_0;
     }
 
-    EvaluateAdaptiveProbeVolume(posWS, normalWS, backNormalWS, viewDir, apvRes,
-        bakeDiffuseLighting, backBakeDiffuseLighting);
+    APVSample apvSample = SampleAPV(posWS, normalWS, viewDir);
+    EvaluateAdaptiveProbeVolume(apvSample, normalWS, backNormalWS, bakeDiffuseLighting, backBakeDiffuseLighting);
 }
 
 void EvaluateAdaptiveProbeVolume(in float3 posWS, in float2 positionSS, out float3 bakeDiffuseLighting)
 {
+    APVResources apvRes = FillAPVResources();
+
     if (_PVSamplingNoise > 0)
     {
         float noise1D_0 = (InterleavedGradientNoise(positionSS, 0) * 2.0f - 1.0f) * _PVSamplingNoise;
         posWS += noise1D_0;
     }
 
-    APVResources apvRes = FillAPVResources();
-    bakeDiffuseLighting = EvaluateAdaptiveProbeVolumeL0(posWS, float3(0.0f, 0.0f, 0.0f), float3(0.0f, 0.0f, 0.0f), apvRes);
+    float3 uvw;
+    if (TryToGetPoolUVW(apvRes, posWS, 0, 0, uvw))
+    {
+        bakeDiffuseLighting = SAMPLE_TEXTURE3D_LOD(apvRes.L0_L1Rx, s_linear_clamp_sampler, uvw, 0).rgb;
+    }
+    else
+    {
+        bakeDiffuseLighting = EvaluateAmbientProbe(0);
+    }
+}
+
+// -------------------------------------------------------------
+// Reflection Probe Normalization functions
+// -------------------------------------------------------------
+
+
+// Same idea as in Rendering of COD:IW [Drobot 2017]
+float GetReflProbeNormalizationFactor(float3 sampleDir, float4 reflProbeSHL0L1, float4 reflProbeSHL2_1, float reflProbeSHL2_2)
```

Comment:
Regarding naming, it is a bit weird
GetReflProbeNormalizationFactor
GetReflectionProbeNormalizationFactor
can't you both call them "GetReflectionProbeNormalizationFactor" ? they have different attribute, 
or maybe one Internal_GetReflectionProbeNormalizationFactor() ?

Reply:
Definitively yes, will rename